src/javascripts/lib/i18n.js

```diff
@@ -0,0 +1,68 @@
+import manifest from '../../manifest.json'
+
+// map to store the key/translation pairs of the loaded language
+let translations
+
+/**
+ * Replace placeholders in the given string with context
+ * @param {String} str string with placeholders to be replaced
+ * @param {Object} context object contains placeholder/value pairs
+ * @return {String} formatted string
+ */
+function parsePlaceholders (str, context) {
+  const regex = /{{(.*?)}}/g
+  const matches = []
+  let match
+
+  do {
+    match = regex.exec(str)
+    if (match) matches.push(match)
+  } while (match)
+
+  return matches.reduce((str, match) => {
+    const newRegex = new RegExp(match[0], 'g')
+    str = str.replace(newRegex, context[match[1]])
```

Comment:
As mentioned during our standup today with @ocke ,

The bug can occur within this line. If we go ` str = str.replace(newRegex, context[match[1]] || '')`, we won't get *undefined* if the value is undefined. 😆

Reply:
Makes sense